go/pkg/trust/signer_gen_no_db.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 ETH Zurich
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package config
+
+package trust
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+
+	"github.com/scionproto/scion/go/lib/addr"
+	"github.com/scionproto/scion/go/lib/scrypto/cppki"
+	"github.com/scionproto/scion/go/lib/serrors"
+	"github.com/scionproto/scion/go/pkg/trust/internal/metrics"
+)
+
+//SignerGenNoDB unlike SignerGen does need a database.
+//Since non CS services might not have a trust database
+//they can query the cs for trc and cert chains and generate signatures for messages.
+//SignerGenNoDB generates signers from the keys available in key dir.
+//It does not require a DB connection. It instead takes in SignedTRC assumed to be active and Chains
+type SignerGenNoDB struct {
```

Comment:
`SignerGenNoDB` should replace `SignerGen` since it is a drop in replacement, and `SignerGen` does not provide any additional feature.

Reply:
SignerGen would be faster because it fetches from a DB and SignerGenNoDB has to make multiple calls to other services. So I think it makes sense to keep SignerGen for services that have access to the database.